pkg/configmap/configmap_test.go

```diff
@@ -196,20 +196,34 @@ func TestDelete(t *testing.T) {
 	}
 }
 
-func TestWithOptions(t *testing.T) {
-	testBuilder := buildTestBuilderWithFakeObjects([]runtime.Object{})
-
-	testBuilder.WithOptions(func(builder *Builder) (*Builder, error) {
-		return builder, nil
-	})
-
-	assert.Equal(t, "", testBuilder.errorMsg)
+func TestUpdate(t *testing.T) {
+	testCases := []struct {
+		testCM        *Builder
+		testData      map[string]string
+		expectedError string
+	}{
+		{
+			testCM:        buildTestBuilderWithFakeObjects([]runtime.Object{}),
+			testData:      map[string]string{"key1": "value1", "key2": "value2"},
+			expectedError: "",
+		},
+		{
+			testCM:        nil,
+			testData:      map[string]string{"key1": "value1", "key2": "value2"},
+			expectedError: "error: received nil ConfigMap builder",
+		},
+	}
 
-	testBuilder.WithOptions(func(builder *Builder) (*Builder, error) {
-		return builder, errors.New("error")
-	})
+	for _, testCase := range testCases {
+		testCase.testCM.WithData(testCase.testData)
+		_, err := testCase.testCM.Update()
 
-	assert.Equal(t, "error", testBuilder.errorMsg)
+		if testCase.expectedError != "" {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		} else {
```

Comment:
Can you change expectedError to an error type using fmt.Errorf and always check it (even if it is nil)?
```
assert.Equal(t, testCase.expectedError, err)
```
Then just check the data if expectedError is nil
```
if testCase.expectedError == nil {
  assert.Equal(t, testCase.testData, testCase.testCM.Definition.Data)
}

Reply:
@trewest I cannot use err nil verification in this case, cause this object is not really exists:
testCM:        buildTestBuilderWithFakeObjects([]runtime.Object{})

If you insist such verification has to be added I'll need to do a full unit-test refactoring

Reply:
Yes, please make sure that the resource actually exists so we can test it thoroughly. Please look at the deployment test as an example - https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/deployment/deployment_test.go#L492